toolchain/check/eval.cpp

```diff
@@ -363,6 +363,9 @@ static auto GetConstantValue(EvalContext& eval_context,
 // If the given instruction is constant, returns its constant value.
 static auto GetConstantValue(EvalContext& eval_context, SemIR::InstId inst_id,
                              Phase* phase) -> SemIR::InstId {
+  if (!inst_id.has_value()) {
```

Comment:
The rest of the PR makes sense on the face of it, but I am wondering about this change - what is it meant to support / what side effect is it achieving?

Reply:
This fixes a crash when importing a `BindSymbolicName` with a `value_id` of `None`. Various of the other `GetConstantValue` overloads already handle an ID of `None`, but this one didn't because we happened to not need it before. This isn't a problem for normal evaluation of `BindSymbolicName` because that hits the `TryEvalTypedInst<SemIR::BindSymbolicName>` specialization and never calls `GetConstantValue` with `InstId::None`.

Reply:
I see, thanks!